RecoEgamma/PhotonIdentification/plugins/PhotonRegressionValueMapProducer.cc

```diff
@@ -0,0 +1,215 @@
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "DataFormats/Common/interface/View.h"
+
+#include "DataFormats/EgammaCandidates/interface/Photon.h"
+#include "DataFormats/PatCandidates/interface/Photon.h"
+
+#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
+
+#include "RecoEcal/EgammaCoreTools/interface/EcalClusterLazyTools.h"
+
+#include <memory>
+#include <vector>
+
+class PhotonRegressionValueMapProducer : public edm::stream::EDProducer<> {
+
+  public:
+  
+  explicit PhotonRegressionValueMapProducer(const edm::ParameterSet&);
+  ~PhotonRegressionValueMapProducer();
+  
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  private:
+  
+  virtual void produce(edm::Event&, const edm::EventSetup&) override;
+
+  void writeValueMap(edm::Event &iEvent,
+		     const edm::Handle<edm::View<reco::Photon> > & handle,
+		     const std::vector<float> & values,
+		     const std::string    & label) const ;
+  
+  // The object that will compute 5x5 quantities  
+  std::unique_ptr<noZS::EcalClusterLazyTools> lazyToolnoZS;
+
+  // for AOD case
+  edm::EDGetTokenT<EcalRecHitCollection> ebReducedRecHitCollection_;
+  edm::EDGetTokenT<EcalRecHitCollection> eeReducedRecHitCollection_;
+  edm::EDGetTokenT<EcalRecHitCollection> esReducedRecHitCollection_;
+  edm::EDGetToken src_;
+
+  // for miniAOD case
+  edm::EDGetTokenT<EcalRecHitCollection> ebReducedRecHitCollectionMiniAOD_;
+  edm::EDGetTokenT<EcalRecHitCollection> eeReducedRecHitCollectionMiniAOD_;
+  edm::EDGetTokenT<EcalRecHitCollection> esReducedRecHitCollectionMiniAOD_;
+  edm::EDGetToken srcMiniAOD_;
+
+  // Cluster shapes
+  constexpr static char phoFull5x5SigmaIPhiIPhi_[] = "phoFull5x5SigmaIPhiIPhi";
+  constexpr static char phoFull5x5SigmaIEtaIPhi_[] = "phoFull5x5SigmaIEtaIPhi";
+  constexpr static char phoFull5x5E2x5Max_[] = "phoFull5x5E2x5Max";
+  constexpr static char phoFull5x5E2x5Left_[] = "phoFull5x5E2x5Left";
+  constexpr static char phoFull5x5E2x5Right_[] = "phoFull5x5E2x5Right";
+  constexpr static char phoFull5x5E2x5Top_[] = "phoFull5x5E2x5Top";
+  constexpr static char phoFull5x5E2x5Bottom_[] = "phoFull5x5E2x5Bottom";
+};
+
+// Cluster shapes
+constexpr char PhotonRegressionValueMapProducer::phoFull5x5SigmaIPhiIPhi_[];
```

Comment:
(to save lines of code)
could these be moved out to an anonymous namespace so that only ```constexpr static char blah[] = "blah";``` is needed?